src/main/scala/aserralle/akka/stream/kcl/scaladsl/KinesisWorkerSource.scala

```diff
@@ -40,8 +42,12 @@ object KinesisWorkerSource {
           val worker = workerBuilder(
             new IRecordProcessorFactory {
               override def createProcessor(): IRecordProcessor =
-                new IRecordProcessor(queue.offer,
-                                     settings.terminateStreamGracePeriod)
+                new IRecordProcessor(
+                  record =>
+                    Await.result(queue.offer(record),
+                                 settings.backpressureTimeout),
```

Comment:
Can you provide a test for the backpressure timeout? Maybe we'll need to capture the `Await.result` error and use it to stop the stream.